workflow/controller/steps.go

```diff
@@ -255,11 +255,11 @@ func (woc *wfOperationCtx) executeStepGroup(stepGroup []wfv1.WorkflowStep, sgNod
 	for _, childNodeID := range node.Children {
 		childNode := woc.wf.Status.Nodes[childNodeID]
 		step := nodeSteps[childNode.Name]
-		if !childNode.Completed() {
+		if !childNode.Fulfilled() {
 			completed = false
-		} else {
+		} else if childNode.Completed() {
```

Comment:
as above - this is change to logic - please double check you intended it

Reply:
This is intended, thanks for asking to make sure.